platform/ios/src/MLNMapView.mm

```diff
@@ -522,6 +522,19 @@ - (instancetype)initWithFrame:(CGRect)frame styleURL:(nullable NSURL *)styleURL
     return self;
 }
 
+- (instancetype)initWithFrame:(CGRect)frame styleJSON:(NSString *)styleJSON
+{
+    if (self = [super initWithFrame:frame])
+    {
+        MLNLogInfo(@"Starting %@ initialization.", NSStringFromClass([self class]));
+        MLNLogDebug(@"Initializing frame: %@ styleJSON: %@", NSStringFromCGRect(frame), styleJSON);
+        [self commonInit];
+        self.mbglMap.getStyle().loadJSON([styleJSON UTF8String]);
```

Comment:
If approach will first load the default style. It is possible that the user will first see the default style flash before this style is finally loaded.